modules/oe_content_call_proposals/oe_content_call_proposals.module

```diff
@@ -0,0 +1,59 @@
+<?php
+
+/**
+ * @file
+ * Primary module hooks for OpenEuropa Call For Proposals content module.
+ */
+
+declare(strict_types = 1);
+
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Datetime\DrupalDateTime;
+use Drupal\Component\Utility\Unicode;
+
+/**
+ * Implements hook_form_BASE_FORM_ID_alter for the node form.
+ */
+function oe_content_call_proposals_form_node_form_alter(array &$form, FormStateInterface $form_state) {
+  /** @var \Drupal\node\NodeInterface $node */
+  $node = $form_state->getBuildInfo()['callback_object']->getEntity();
+  if ($node->bundle() !== 'oe_call_proposals') {
+    return;
+  }
+
+  $form['oe_call_proposals_deadline']['widget']['#states'] = [
+    'invisible' => [
+      'select[name="oe_call_proposals_model"]' => ['value' => 'permanent']
+    ],
+  ];
+
+  $form['#validate'][] = 'oe_content_call_proposals_node_validate';
+}
+
+/**
+ * Custom validator for the Call For Proposals node form.
+ *
+ * @param array $form
+ *   The form.
+ * @param \Drupal\Core\Form\FormStateInterface $form_state
+ *   The form state.
+ */
+function oe_content_call_proposals_node_validate(array $form, FormStateInterface $form_state) {
```

Comment:
Instead of form validation I suggest to use Constraint validators. See build/modules/contrib/oe_content/modules/oe_content_event/src/Plugin/Validation/Constraint/EventFieldsRequiredValidator.php. Might be build/core/lib/Drupal/Core/Validation/Plugin/Validation/Constraint/LengthConstraint.php can be helpful.
As I know it will be helpful when site will work as a service using API.

Reply:
@yenyasinn I implemented the Constraint validators, can you please review them?